### PR TITLE
[007] Add tasks for homework 7 (collections)

### PR DESCRIPTION
**Pull request by AlexiaSilver**

### CODE REVIEW

File: src/com/epam/javacourse/homework7/StringOperations.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.epam.javacourse.homework7;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.*;
```
Comment (knadonenkoepam) [inline src/com/epam/javacourse/homework7/StringOperations.java]: Wildcard imports are bad :)

File: src/com/epam/javacourse/homework7/StringOperations.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                lines.add(line);
+            }
+        }
+        Collections.sort(lines, Comparator.comparingInt(String::length));
+        return lines;
+    }
+    public static boolean checkParentheses(String s) {
+        Stack<Character> stack = new Stack<>();
+        for (char c : s.toCharArray()) {
+            if (c == '(') {
```
Comment (knadonenkoepam) [inline src/com/epam/javacourse/homework7/StringOperations.java:27]: Just for the future check this pattern https://medium.com/swlh/return-early-pattern-3d18a41bba8
Don't change anything right now :)

File: src/com/epam/javacourse/homework7/Main.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
+package com.epam.javacourse.homework7;
+
+import java.io.IOException;
+import java.util.*;
+
+public class Main {
+    public static void main(String[] args) {
+        String inputFilePath = "src/com/epam/javacourse/resources/poem.txt";
```
Comment (knadonenkoepam) [inline src/com/epam/javacourse/homework7/Main.java]: It's good practice to keep this parameters as final if you don't going to change them in future

File: src/com/epam/javacourse/homework7/Main.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        // Reverse the contents of a file (1)
+        try {
+            FileOperations.reverseFile(inputFilePath, outputFilePath);
+        } catch (IOException e) {
+            System.out.println("An error occurred while reversing the file: " + e.getMessage());
+        }
+
+        //Sort the lines by length (2)
+        try {
+            List<String> sortedLines = StringOperations.sortLinesByLength("src/com/epam/javacourse/resources/poem.txt");
```
Comment (knadonenkoepam) [inline src/com/epam/javacourse/homework7/Main.java]: And that's a duplication, you have a string parameter

**Review by knadonenkoepam — DISMISSED** [review]
Overall a good work

File: src/com/epam/javacourse/homework7/StringOperations.java (hunk 1 above)
Comment (AlexiaSilver) [reply src/com/epam/javacourse/homework7/StringOperations.java]: Fixed

File: src/com/epam/javacourse/homework7/StringOperations.java (hunk 2 above)
Comment (AlexiaSilver) [reply src/com/epam/javacourse/homework7/StringOperations.java:27]: Got your point:)

File: src/com/epam/javacourse/homework7/Main.java (hunk 3 above)
Comment (AlexiaSilver) [reply src/com/epam/javacourse/homework7/Main.java]: Fixed

File: src/com/epam/javacourse/homework7/Main.java (hunk 4 above)
Comment (AlexiaSilver) [reply src/com/epam/javacourse/homework7/Main.java]: Fixed

File: src/com/epam/javacourse/homework7/Main.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
+package com.epam.javacourse.homework7;
+
+import java.io.IOException;
+import java.util.*;
+
+public class Main {
+    public static void main(String[] args) {
+        final String inputFilePath = "src/com/epam/javacourse/resources/poem.txt";
+        String outputFilePath = "src/com/epam/javacourse/resources/poem.txt";
```
Comment (knadonenkoepam) [inline src/com/epam/javacourse/homework7/Main.java]: This one should be final too, and I think you don't need this parameter as it's the same with inputFilePath

Comment (AlexiaSilver) [reply src/com/epam/javacourse/homework7/Main.java]: Fixed

**Review by knadonenkoepam — APPROVED** [review]

**Pull request merged**